caddyconfig/httpcaddyfile/httptype.go

```diff
@@ -719,8 +719,12 @@ func (st *ServerType) serversFromPairings(
 				// If TLS is specified as directive, it will also result in 1 or more connection policy being created
 				// Thus, catch-all address with non-standard port, e.g. :8443, can have TLS enabled without
 				// specifying prefix "https://"
+				// Second part of the condition is to allow creating TLS conn policy even though `auto_https` has been disabled
+				// ensuring compability with behavior described in below link
```

Comment:
```suggestion
				// ensuring compatibility with behavior described in below link
```

Reply:
Yes, I've fixed this in latest commit ;)
Nice catch though!